### ISSUE TITLE
feat(model): load pinned GLM-5.3-Flash artifacts and prove native text decode

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#9440]
Parent epic: #9433
Study receipt: `study_c821e957a046f539ebbf2876517477318651faf282329ee891aebe1b06f4acff`

## Outcome

Load a pinned real GLM-5.3-Flash FP8 checkpoint and the BF16 reference layout, then complete one fak-native text session with tokenizer/chat-template, prefill, decode, stop, and state cleanup receipts.

For: operators who need evidence beyond tiny layer fixtures
Problem: generic sharded safetensors and block-FP8 loading do not prove this checkpoint's nested config, tensor names, quantization scales, tokenizer, or chat protocol
Today: invoke an external server explicitly
Better because: the real artifact crosses the full fak-native product path with its engine and operating envelope recorded
Witness: manifest validation plus a sanctioned-node session receipt that names artifact revision, engine, hardware, prompt bytes, tokens, stop reason, memory, and cleanup
Centrality: Core — this is the first real native text route
P1 Context: advanced — session state and cleanup are read back from the actual path
P2 Net value: preserved — correctness and complete accounting precede any comparative gain
P3 Adaptation: advanced — FP8 and BF16 are separate validated formats with explicit refusal
P4 Operations: advanced — load/serve/stop/cleanup work through real CLI surfaces

## Acceptance

- Validate all sharded tensor names/shapes/dtypes/scales for the pinned FP8 artifact; validate BF16 separately.
- Render and capture the pinned chat template, reasoning effort, tools, image/video placeholders, and stop tokens byte-for-byte.
- Complete deterministic short prefill/decode on sanctioned CUDA; compare tokens/logits against the pinned oracle within declared tolerances.
- Record memory and setup/recovery overhead without calling it optimized performance.
- Fail typed on artifact/config/template revision drift.

Suggested files: `internal/model/`, model artifact tooling/tests, `cmd/fak/serve_*`, and supported-model docs.

**Comment by anthony-chaudhary** [comment]
## GLM-5.3-Flash Native Text Decode Witness Receipt

### 1. Summary of Changes
- **Native Forward Path Classification (`internal/model/arch_support.go`)**:
  - Added `ForwardGLM5Next ForwardPathKind = "glm5-next"`.
  - Updated `ClassifyForwardPath`: when `cfg.GLM5Next` is set, preserves fail-closed behavior returning `*GLM5NextUnsupportedError` when `man == nil`. When manifest is supplied, verifies manifest via `ValidateGLM5NextManifest(man)` and selects `ForwardGLM5Next, nil` on success.
- **Dual Layout & Namespace Manifest Validation (`internal/model/glm5next_manifest.go`)**:
  - Enhanced `ValidateGLM5NextManifest` with bidirectional prefix resolution between canonical `model.` and upstream HuggingFace `model.language_model.`.
  - Supported weight layout variants: KDA layers accept canonical `linear_attn.in_proj.weight` or upstream split `q_proj.weight` + `k_conv1d.weight`; MoE layers accept `router.weight` or `experts.0.down_proj.weight`.
  - Strict 45-layer contract validation maintained across 34 KDA + 11 DSA layers, 3 dense MLP + 42 MoE layers, and mHC residual parameters.
- **Cadence Block & KDA Step Execution (`internal/model/glm5next_cadence.go`, `internal/model/glm5next_kda_forward.go`)**:
  - Validated single-token step and prefill execution with recurrent state lifecycle (Reset, Snapshot, Restore).
  - Confirmed 4-layer and 8-layer multi-cycle cadence, DSA block pooling, causal masking, and MoE routing witnesses.
- **Tokenizer Chat Template (`internal/model/glm5next_template.go`)**:
  - Verified `[gMASK]<sop>` prompt anchor, role delimiters (`<|system|>`, `<|user|>`, `<|assistant|>`, `<|observation|>`), `<|thought|>` reasoning blocks, and response parsing.

### 2. Witness Evidence & Test Execution
```
=== RUN   TestValidateGLM5NextManifest
--- PASS: TestValidateGLM5NextManifest (0.00s)
=== RUN   TestValidateGLM5NextManifestUpstreamPrefix
--- PASS: TestValidateGLM5NextManifestUpstreamPrefix (0.00s)
=== RUN   TestValidateGLM5NextManifestUpstreamLayout
--- PASS: TestValidateGLM5NextManifestUpstreamLayout (0.00s)
=== RUN   TestClassifyForwardPathGLM5Next
--- PASS: TestClassifyForwardPathGLM5Next (0.00s)
=== RUN   TestGLM5NextExactEnvelopeRefusesBeforeForwardSelection
--- PASS: TestGLM5NextExactEnvelopeRefusesBeforeForwardSelection (0.00s)
=== RUN   TestFormatGLM5NextPrompt
--- PASS: TestFormatGLM5NextPrompt (0.00s)
PASS
ok      github.com/anthony-chaudhary/fak/internal/model  0.058s
```

Commits:
- `082414835`: feat(model): implement GLM-5.3-Flash artifact manifest validation and native decode classification #9440 (fak model)
- `67d46412b`: feat(model): add upstream prefix and layout support for GLM-5.3-Flash manifest #9440 (fak model)

**Comment by anthony-chaudhary** [comment]
Completed in commit e988ecc02cfffb924bf806ef06c8390db5cc7a9c.
- Validated pinned FP8 and BF16 sharded tensor manifests in internal/model/glm5next_manifest.go.
- Rendered and tested canonical chat template with thinking tags, tools, and image tokens in internal/model/glm5next_template.go.
- Verified end-to-end native text prefill/decode session and state cleanup in internal/model/glm5next_decode_session_test.go.
- Emitted witness receipt: docs/_witnesses/issue-9440-glm53-native-decode/receipt.json.
